yang/frr-staticd.yang

```diff
@@ -0,0 +1,141 @@
+module frr-staticd {
+  yang-version "1.1";
+  namespace "http://frrouting.org/yang/staticd";
+
+  prefix frr-staticd;
+
+  import frr-routing {
+    prefix "frr-rt";
+  }
+
+  import frr-nexthop {
+    prefix frr-nexthop;
+  }
+
+  import ietf-inet-types {
+    prefix inet;
+  }
+
+  organization
+    "Free Range Routing";
+
+  contact
+    "FRR Users List: <mailto:frog@lists.frrouting.org>
+     FRR Development List: <mailto:dev@lists.frrouting.org>";
+
+  description
+    "This module defines a model for managing FRR staticd information.
+     This YANG module augments the ietf-routing with additional
+     nexthop information";
+
+  revision 2019-12-03 {
+    description
+      "Initial revision.";
+  }
+
+
+  augment "/frr-rt:routing/frr-rt:control-plane-protocols/"
+        + "frr-rt:control-plane-protocol" {
+    container staticd {
+      when "../frr-rt:type = 'frr-rt:static'" {
+        description
+          "This container is only valid for the 'static' routing
+           protocol.";
+      }
+      description
+      "Support for a 'static' pseudo-protocol instance
+       consists of a list of routes.";
+
+      list address-family {
+        key "address-family";
+        description
+          "Each list entry for one address family.";
+        uses frr-rt:address-family;
+        list ipv4-route {
+          when "../address-family = 'frr-rt:ipv4'" {
+            description
+              "Only applicable to IPv4 address family.";
+          }
+          key "destination-prefix";
+          description
+            "A list of static routes.";
+
+          leaf destination-prefix {
+            type inet:ipv4-prefix;
+            mandatory true;
+            description
+            "IPv4 destination prefix.";
+          }
+          container frr-staticd-next-hop {
+            description
+              "Augment static route configuration 'nexthop-list'";
+            uses frr-nexthop:frr-nexthop;
+          }
+        }
+        list ipv6-route {
+          when "../address-family = 'frr-rt:ipv6'" {
+            description
+              "Only applicable to IPv6 address family.";
+          }
+          key "destination-prefix";
+          description
+            "A list of static routes.";
+
+          leaf destination-prefix {
+            type inet:ipv6-prefix;
+            mandatory true;
+            description
+              "IPv6 destination prefix.";
+          }
+
+          container frr-staticd-next-hop {
+            description
+              "Augment static route configuration 'nexthop-list'";
+            uses frr-nexthop:frr-nexthop;
+          }
+        }
+      }
+    }
+  }
+
+  augment "/frr-rt:routing/frr-rt:control-plane-protocols/"
+        + "frr-rt:control-plane-protocol/staticd/address-family/"
+        + "ipv4-route/frr-staticd-next-hop/frr-nexthops/nexthops" {
+
+    leaf distance {
```

Comment:
I think both distance and tag are properties of a static route, not of each nexthop in a route? check `lib/zclient.h` to see how those attributes are conveyed.

Reply:
Actually at present in staticd, if you go to the static_route struct, then tag and distance are defined in this struct means part of nexthop. Now when route goes to zebra then these attributes becomes part of route.